usecase/pet.go

```diff
@@ -21,17 +21,28 @@ func (c *PetUseCase) FindById(userID uniqueEntity.ID) (*entity.Pet, error) {
 	return nil, nil
 }
 
-func (c *PetUseCase) Update(petID string, userID string, petToUpdate *entity.Pet) (err error) {
+func (petUseCase *PetUseCase) Update(petID string, userID string, petToUpdate *entity.Pet) (err error) {
 
 	updateValues := map[string]interface{}{}
 
-	if c.isValidPetSize(petToUpdate) {
-		updateValues["size"] = &petToUpdate.Size
+	if petUseCase.isValidPetSize(petToUpdate) {
+		updateValues["size"] = petToUpdate.Size
 	} else {
 		return errors.New("the animal size is invalid")
 	}
 
-	err = c.repo.Update(petID, userID, updateValues)
+	if petUseCase.isValidWeight(petToUpdate) {
+		updateValues["weight"] = petToUpdate.Weight
+		updateValues["weight_measure"] = petToUpdate.WeightMeasure
+	} else {
+		return errors.New("the animal weight is invalid")
```

Comment:
Acredito que não precisaria retorna um erro caso o weight não seja informado no body. Pela rota ser um PATCH, pode atualizar 1 ou mais campos

Reply:
Mas tem que avisar se o peso for enviado e for invalido